### ISSUE TITLE
Ability to update to latest set of Anaconda packages

### ISSUE THREAD

**Issue by ijstokes** [ContinuumIO/anaconda-issues#1303]
I think a lot of people want to be able to do what I described in this ML response:

https://groups.google.com/a/continuum.io/forum/#!msg/anaconda/Q50Yl1V2q9g/yNdqhll4CQAJ

I certainly have wanted to be able to do this, and I'm happy that I've now figured out a way to do it.  This is **probably** something that needs to involve the Conda team, but the main driver for it today is, I think, having this capability for Anaconda and available in the Anaconda distribution.  IOW, the ability for a user to say "give me the latest version of all the Anaconda packages".  I don't think we have an easy way for someone to do that today, but we should.

**Comment by mingwandroid** [comment]
Good work. One thing though, you said in that reply:

> And now with the Conda-Forge community group a large set (>1000) additional packages are also rapidly built and released through them (no more difficult than i) deciding you trust them to build software that will run on your machine; and ii) adding the `conda-forge` channel to your `.condarc` or explicitly doing `-c conda-forge` on each `conda install` or `conda update` command).

I would strongly encourage you not to mention that people should try to use software from both `conda defaults` and `conda-forge` in the same conda environment. We are binary incompatible. I spend a large amount of time apologizing to people and trying to guide them when the have broken their environments by doing this.

If people want to use `conda-forge` they should see it as a choice to use that instead of `conda defaults`.

**Comment by ijstokes** [comment]
@mingwandroid: yes, I forget about the complications that are introduced from having both `defaults` and `conda-forge` in your `.condarc`.  Do we know if `conda-forge` now builds ~all of the Anaconda packages?  I would guess that it does not and if not I don't see how it could be possible to remove `defaults` and **just** use `conda-forge`.

I know there has been a lot of internal discussion on the impact of Conda-Forge.  I haven't been a part of those conversations or had time to read notes/outcomes.

**Comment by mingwandroid** [comment]
They don't build everything but they aim to. AFAIK, once that happens they will look to stop bootstrapping using `Miniconda` and will release their own installers.

`conda-forge` software can run on `conda defaults` software up to the point when packages for those dependencies get added to `conda-forge` itself. The reverse is not true.

Therefore, by putting `conda-forge` first in your `channels` in `~/.condarc` you should be able to more or less guarantee that things available from both get pulled from `conda-forge`.

**Comment by ijstokes** [comment]
@kalefranz this is the thing we were just talking about (and the linked ML response).

**Comment by ijstokes** [comment]
@kalefranz rightly pointed out that a problem with this strategy is that many packages do not specify upper bounds on their dependencies (because it can be hard to specify *a priori* when future versions of the dependency will break the package in question) and therefore implicitly will accept **any** version of a dependency package later than some minimum.  But then a "newer" version of the dependency comes along and it **does** break the package in question but Conda doesn't know about this.  Assuming "semver" might help and then some stricter packaging constraints, but that introduces other shortcomings.
